src/Service.Tests/CosmosTests/QueryTests.cs

```diff
@@ -472,6 +479,158 @@ ... p
+                                            query: graphQLQuery,
+                                            queryName: queryName
+                                            );
+                name = queryResponse.GetProperty("name").GetString();
+
+                var update = new
+                {
+                    id = id,
+                    name = "new_name",
+                };
+
+                // Mutated name property
+                JsonElement mutationResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                client,
+                                server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                query: mutation,
+                                queryName: "updatePlanet",
+                                variables: new() { { "id", id }, { "partitionKeyValue", id }, { "item", update } },
+                                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(),
+                                clientRoleHeader: AuthorizationResolver.ROLE_AUTHENTICATED
+                                );
+
+                // Asserting name is mutated
+                Assert.IsFalse(queryResponse.GetProperty("name").GetString().Equals(update.name), "Mutation didn't change the name successfully");
+
+                // Second query - the data returned is from the cache, not the mutated one
+                JsonElement queryResponse2 = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                             client,
+                                             server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                             query: graphQLQuery,
+                                             queryName: queryName
+                                             );
+
+                // Asserting cached data is returned
+                Assert.IsFalse(queryResponse.GetProperty("name").GetString() != name, "Query didn't return cached value");
+
+                update = new
+                {
+                    id = id,
+                    name = name,
+                };
+
+                // Clean up setting document name back to original value
+                _ = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                client,
+                                server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                query: mutation,
+                                queryName: "updatePlanet",
+                                variables: new() { { "id", id }, { "partitionKeyValue", id }, { "item", update } },
+                                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(),
+                                clientRoleHeader: AuthorizationResolver.ROLE_AUTHENTICATED
+                                );
```

Comment:
Right. I have removed the cleanup that I was performing in my test. And yes, it makes sense to revisit the TestCleanUp function, which is creating the container multiple times. Perhaps we can perform cleanup at the assembly level instead for each test. This will require some analysis. I'll create a separate item to check on this. Additionally, we are inquiring with the Emulator team about the issues we are facing in Linux and macOS images. Once those are resolved, we can fully transition to the emulator. However, we can start looking into this before the emulator team provides a fix.

Reply:
please comment here the link to the issue created for this task.

Reply:
https://github.com/Azure/data-api-builder/issues/2082